test/sampling_benchmark.cpp

```diff
@@ -12,13 +12,124 @@
 // Our working directory is generators/build so one up puts us in the root directory:
 #define MODEL_PATH "../../test/test_models/"
 
+extern std::unique_ptr<OrtEnv> g_ort_env;
+
+// Defined in sampling_tests.cpp
+void CreateRandomLogits(float* logits, int num_large, int vocab_size, int batch_size, std::mt19937& engine);
+
+TEST(Benchmarks, BenchmarkRandomizedSamplingTopPCpu) {
```

Comment:
we can set up a microbenchmark with google's microbenchmark library: https://github.com/google/benchmark